File: lib/hw3/Homework3_1.rb
# Bulls and Cows
require 'action_pack'
require 'action_view'
require 'pry'

class Gamemaster
  include ActionView::Helpers::TextHelper
  def initialize
    pick_numbers
    @solved = false
    @guesses = 0
  end

  def solved?
    @solved
  end

  def pick_numbers
    @numbers = []
    4.times do
      @numbers << rand(0..9).to_s
    end
  end

  def check(guess)
    @guesses += 1
    guess = guess.split("")
    throw IncorrectGuess if guess.size != 4
    bulls = 0
    cows = 0
    guess.each_with_index do |num, index|
      # binding.pry
      if @numbers[index] == num
        bulls += 1
      elsif @numbers.include?(num)
        cows += 1
      end
    end
    if bulls == 4
      @solved = true
      puts "You guessed it!  And in only #{pluralize(@guesses, 'guess')}!  What a pro."
    else
      puts "#{pluralize(bulls, 'bull')} and #{pluralize(cows, 'cow')}"
    end
  end 
end

gm = Gamemaster.new

while !gm.solved? do
  puts "Enter your guess"
  guess = gets.chomp
  gm.check(guess)
end

File: turnin/hw3/Homework3_Cosgrove/Homework3_1.cpp
/*
Will Cosgrove
121007913
will@willcosgrove.com
Section 509
2/14/2012
Homework 3, Problem 1
*/

#include "std_lib_facilities.h"

class Gamemaster {
  int guesses;
public:
  bool solved;
  vector<int> numbers;
  Gamemaster(){
    pick_numbers();
    guesses = 0;
    solved = false;
  }

  void pick_numbers(){
    for(int i = 0; i < 4; i++){
      numbers.push_back(rand() % 10);
    }
  }

  void check(string guess){
    guesses++;
    int bulls = 0;
    int kine = 0;
    for(int i = 0; i < 4; i++){
      int current_number = guess[i] - 48;
      if(numbers[i] == current_number){
        bulls++;
      } else if(find(numbers.begin(), numbers.end(), current_number) != numbers.end()){
        kine++;
      }
    }
    if(bulls == 4){
      solved = true;
      cout << "You solved it!  And in only " << guesses << " guesses!  What a pro.\n";
    } else {
      cout << bulls << " bulls " << kine << " kine\n";
    }
  }
};

int main(){
  string guess;
  Gamemaster g = Gamemaster(); // Initialize the Gamemaster
  while(!g.solved){ // While the game is not solved
    cout << "Make a guess\n";
    cin >> guess;
    g.check(guess); // Check for bulls and kine
  }
  return 0;
}